src/utils/index.ts

```diff
@@ -191,5 +233,5 @@ export function getTokenDisplay(token: Token, chainId: ChainId): string {
 }
 
 export function isTokenXDAI(tokenAddress?: string, chainId?: ChainId): boolean {
-  return !!tokenAddress && !!chainId && tokenAddress == WETH[chainId].address && chainId == 100
+  return !!tokenAddress && !!chainId && chainId == 100 && tokenAddress == WETH[100].address
```

Comment:
This was giving an error with WETH coming from a  library with no support for polygon, because before the number of the chain is also checked there was no need to pass `WETH[chainId]`

Reply:
For this, we need to clone the library and publish our own one. Could you take care of it?

Reply:
Or maybe there is already a clone that works well with xdai polygon and bsc

Reply:
@josojo Tried with this fork https://github.com/protofire/uniswap-sdk/tree/support-bsc-polygon but for some reason I can't make it work with the time left.

Reply:
Hm... why didn't you fork from the working lib here:
https://github.com/gnosis/ido-ux/blob/develop/package.json#L143
?